postgres/server/init.sls

```diff
@@ -192,3 +192,25 @@ postgresql-running:
       - file: postgresql-pg_ident
 
 {%- endif %}
+
+{%- if postgres.port|default(false) %}
+
+postgresql-port:
+  file.replace:
+    - name: {{ postgres.conf_dir }}/postgresql.conf
+    - pattern: ^#*\s*(port)\s*=\s*\d{4,5}(\s*).*$
+    - repl: >-
+        \1 = {{ postgres.port }}\2# Managed by SaltStack: please do not edit
+    - flags: 8  # ['MULTILINE']
+    - show_changes: True
+    - append_if_not_found: True
+    - backup: {{ postgres.config_backup|default(false, true) }}
+    - require:
+      - file: postgresql-config-dir
+    - watch_in:
+       - service: postgresql-port
+
+  service.running:
+    - name: {{ postgres.service }}
```

Comment:
I think having two states handling a single service is not that good idea, since this could be very easy overlooked in further development...

But I understand we'd have to deal with `port` setting could be defined outside of `Managed by SaltStack` protected zone. So we can add a `file.comment` state with rather simple regex to force commenting out string matching `/port = .+/` **only when** `postgres.port` is defined and there would be changes in `postgresql-conf`:

```yaml
{%- if salt['pillar.get']('postgres.port') %}

postgresql-alter-port:
  file.comment:
    - name: {{ postgres.conf_dir }}/postgresql.conf
    - regex: ^port = .+
    - prereq:
      - file: postgresql-conf

{%- endif %}
```